### PR TITLE
refactor(semantic): simplify raising errors in transform checker

### PR DESCRIPTION
**Pull request by overlookmotel**
Simplify creating errors in transformer checker.

### CODE REVIEW

**Comment by overlookmotel** [comment]
* **#5051** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5051?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by overlookmotel** [comment]
I [said I'd finished with refactoring](https://github.com/oxc-project/oxc/pull/5050#issuecomment-2302521598). I lied! But *now* I have...

File: crates/oxc_semantic/src/post_transform_checker.rs [hunk 1]
```diff
@@ -163,10 +162,27 @@ struct SemanticData<'s> {
...
 
+#[derive(Default)]
+struct Errors(Vec<OxcDiagnostic>);
+
+impl Errors {
+    fn push<S: AsRef<str>>(&mut self, message: S) {
+        self.0.push(OxcDiagnostic::error(message.as_ref().trim().to_string()));
+    }
+
+    fn get(self) -> Option<Vec<OxcDiagnostic>> {
```
Comment (DonIsaac) [inline crates/oxc_semantic/src/post_transform_checker.rs:173]: Maybe make this `impl From<Errors> for Option<Vec<OxcDiagnostic>>`?

Comment (Dunqing) [reply crates/oxc_semantic/src/post_transform_checker.rs:173]: The stack is huge, let's merge it first

**Pull request merged**